tests/test_http.py

```diff
@@ -182,6 +182,57 @@ def test_size_exceeded(self):
                     b"",
                 )
 
+    def test_size_check_ignore_files(self):
+        body = (
+            b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="title"\r\n\r\n'
+            + b"My First Book\r\n"
+            + b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="pdf"; filename="book.pdf"\r\n\r\n'
+            + b"FAKEPDFBYTESGOHERETHISISREALLYLONGBUTNOTUSEDTOCOMPUTETHESIZEOFTHEREQUEST"
+            + b"--BOUNDARY--"
+        )
+        request = AsgiRequest(
+            {
+                "http_version": "1.1",
+                "method": "POST",
+                "path": "/test/",
+                "headers": {
+                    "content-type": b"multipart/form-data; boundary=BOUNDARY",
+                    "content-length": str(len(body)).encode("ascii"),
+                },
+            },
+            body,
+        )
+        with override_settings(DATA_UPLOAD_MAX_MEMORY_SIZE=60):
+            request.POST
+
+    def test_size_check_ignore_files_but_honor_other_post_data(self):
+        body = (
+            b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="title"\r\n\r\n'
+            + b"My First Book\r\n"
+            + b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="pdf"; filename="book.pdf"\r\n\r\n'
+            + b"FAKEPDFBYTESGOHERETHISISREALLYLONGBUTNOTUSEDTOCOMPUTETHESIZEOFTHEREQUEST"
+            + b"--BOUNDARY--"
+        )
+        request = AsgiRequest(
+            {
+                "http_version": "1.1",
+                "method": "POST",
+                "path": "/test/",
+                "headers": {
+                    "content-type": b"multipart/form-data; boundary=BOUNDARY",
+                    "content-length": str(len(body)).encode("ascii"),
+                },
+            },
+            body,
+        )
```

Comment:
From 211 to here seems like the same code as from 186 to 206, a little refactor will be required prior to merging.

Reply:
Both body are different, but we could probably factorize the `AsgiReqest` part?

```
def MultipartAsgiRequest(body, **kwargs):
    data = {
        "http_version": "1.1",
        "method": "POST",
        "path": "/test/",
        "headers": {
            "content-type": b"multipart/form-data; boundary=BOUNDARY",
            "content-length": str(len(body)).encode("ascii"),
        },
    }
    data.update(kwargs)
    return AsgiRequest(data, body)
```

Then, in tests:

```
    def test_size_check_ignore_files_but_honor_other_post_data(self):
        body = (
            b"--BOUNDARY\r\n"
            + b'Content-Disposition: form-data; name="title"\r\n\r\n'
            + b"My First Book\r\n"
            + b"--BOUNDARY\r\n"
            + b'Content-Disposition: form-data; name="pdf"; filename="book.pdf"\r\n\r\n'
            + b"FAKEPDFBYTESGOHERETHISISREALLYLONGBUTNOTUSEDTOCOMPUTETHESIZEOFTHEREQUEST"
            + b"--BOUNDARY--"
        )
        request = MultipartAsgiRequest(body)
```
Would that be acceptable to you?

Reply:
Sure, then perhaps put the boundary in a module-level or class-level variable so that you can reuse it in both test without having to duplicate the definition ? Or is there a difference between the boundary of both tests that I didn't spot ?

Reply:
Very nice contribution Eliot, thank you very much for fixing this ! Once the copy/pasted code is refactored into reused code I will approve this PR.